### ISSUE TITLE
Weird CRS

### ISSUE THREAD

**Issue by kravietz** [patrickbwarren/qgis3-survex-import#2]
I have just tried to import the 3D file generated from Therion and it works except the shape lands on completely different place on the Earth. The original location of the entrance is specified using these commands in Therion:
```
cs lat-long
fix 0 43.35333 43.40117 1397
```
The KML seems to be placed correctly, both when loaded into QGIS and Google Earth.

[su-2.zip](https://github.com/patrickbwarren/qgis3-survex-import/files/4054891/su-2.zip)

**Comment by patrickbwarren** [comment]
Hi,
When I do dump3d on the file you sent, I get

TITLE "su-2.3d"
DATE "@1578924130"
DATE_NUMERIC 1578924130
CS +proj=utm +zone=38 +ellps=WGS84 +datum=WGS84 +units=m
 [ ... ]
NODE 370430.21 4801293.93 1397.00 [Suu-Akan-2.0] UNDERGROUND ENTRANCE FIXED
[ ... ]

Indeed WGS84 lat-long 43.35333 N, 43.40117 E comes out to be UTM (WGS84) 38
N 370430 4801294, so it seems that Therion reprojects to UTM to create the
.3d file with a metric co-ordinate system.

When I tried to import this seems to end up in EPSG:4326 not EPSG:32638 as
it should.  I don't know why this is - there is a complicated interplay
between converting the proj4 string in the .3d file, to the expected EPSG
number.

There is a workaround: set the project CRS to EPSG:32638 which is UTM zone
38N, then import the .3d file with "CRS from QGIS project" selected.  The
data seems to land in the right CRS.

Patrick

**Comment by patrickbwarren** [comment]
More comments:

1.The log file shows the ImportFromProj4(string) is converting "+proj=utm +zone=38 +ellps=WGS84 +datum=WGS84 +units=m"  to EPSG:4326 which latter is WGS84 long/lat not UTM.  So it could be that this is a malformed proj4 string that Therion is generating.  Checking in QGIS, for WGS84 UTM zone 38N (EPSG:32638) the proj4 string should be "+proj=utm +zone=38 +datum=WGS84 +units=m +no_defs" (not the +ellps=WGS84 is missing compared to Therion).  I will investigate further when I have time.

2. Even so, when I deselect the two CRS checkboxes in the import dialog, the log shows that the memory layers are correctly being created _without_ specifying a CRS.  So the fact that these land with the wrong CRS is _not_ a problem with the import plugin, rather it is due to a complicated interplay between the vector layer creation defaults, the project CRS, the project defaults, and the user profile defaults.  Somehow one can probably get QGIS to do the 'right thing' and open a CRS dialog box for the user to specify the CRS when a memory layer is created without specifying a CRS.

**Comment by patrickbwarren** [comment]
I'm treating this as a _bug_ - now fixed in current snapshot.  I have improved the CRS selection mechanism so that proj4 strings are handled correctly, including therion generated ones; and a CRS selector dialog box appears if the two CRS checkboxes in the import dialog are both deselected (doing the right thing).  Under the hood, layer.setCrs() is invoked with the selected CRS, rather than creating layers with an EPSG code.  This avoids loss of information in extracting an EPSG code from a random proj4 string.  I have now tested this with quite a large selection of .3d files and it all works.  It only remains update the documentation and make a new release, then close this issue.
